right/src/slave_drivers/uhk_module_driver.c

```diff
@@ -73,6 +75,25 @@ void UhkModuleSlaveDriver_Init(uint8_t uhkModuleDriverId)
     uhkModuleState->pointerDelta.y = 0;
 }
 
+// When module is swapped, we need to reload its Keymap once we know its
+// moduleId. However, this triggers macro events, and we don't want to trigger
+// the same macro multiple times if we can detect that another module is also
+// going through the initialization sequence.
+void ReloadKeymapIfNeeded()
+{
+    bool someoneElseWillDoTheJob = false;
+    for (uint8_t i = 0; i < UHK_MODULE_MAX_SLOT_COUNT; i++) {
+        uhk_module_state_t *uhkModuleState = UhkModuleStates + i;
+        uhk_slave_t* slave = Slaves + i;
+
+        someoneElseWillDoTheJob = uhkModuleState->moduleId == 0 && slave->isConnected;
```

Comment:
What's the logic behind this line? moduleId==0, hence the module is disconnected, but the slave is connected.

Reply:
If I read the code correctly, then `isConneted` is set to true after first successfull transfer. However, moduleId is not known 
until roughly 4th exchange - that is after 8th transfers are finished. 

This means that after startup, or whenever left half (with a dangling keycluster module) is reconnected, the modules will go through their initiation sequences in parallel. I.e., being connected, obviously knowing slot id, but not yet knowing the module id.